crates/ntx-builder/src/state/tests.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::HashSet;
+
+use miden_node_proto::domain::mempool::MempoolEvent;
+use miden_node_utils::fee::test_fee_params;
+use miden_objects::Word;
+use miden_objects::block::BlockHeader;
+use miden_objects::crypto::merkle::{MmrPeaks, PartialMmr};
+use miden_objects::note::Nullifier;
+use miden_objects::transaction::{PartialBlockchain, TransactionId};
+
+use crate::state::State;
+use crate::store::StoreClient;
+
+/// Helper function to create a mock State for testing without needing a real store.
+fn create_mock_state() -> State {
+    // Create a minimal genesis block header
+    let chain_tip_header = BlockHeader::new(
+        1_u8.into(),       // version
+        Word::default(),   // prev_hash
+        0_u32.into(),      // block_num (genesis)
+        Word::default(),   // chain_root
+        Word::default(),   // account_root
+        Word::default(),   // nullifier_root
+        Word::default(),   // note_root
+        Word::default(),   // tx_hash
+        Word::default(),   // kernel_root
+        Word::default(),   // proof_hash
+        test_fee_params(), // fee_parameters
+        0_u32,             // timestamp
+    );
+
+    // Create an empty partial blockchain
+    let chain_mmr = PartialBlockchain::new(PartialMmr::from_peaks(MmrPeaks::default()), Vec::new())
+        .expect("Failed to create empty PartialBlockchain");
```

Comment:
I'm surprised there isn't an empty method?

Reply:
`PartialBlockchain::default` exists (https://docs.rs/miden-objects/latest/miden_objects/transaction/struct.PartialBlockchain.html#impl-Default-for-PartialBlockchain)